python/ray/_private/test_utils.py

```diff
@@ -2158,3 +2159,107 @@ def get_ray_default_worker_file_path():
         f"/home/ray/anaconda3/lib/python{py_version}/"
         "site-packages/ray/_private/workers/default_worker.py"
     )
+
+
+class Cgroup2NetworkBlocker:
```

Comment:
This is really cool. If it works, I think it'll be helpful for testing other network failure cases.